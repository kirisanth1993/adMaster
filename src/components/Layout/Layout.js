import './Layout.scss';
import { Route } from "react-router-dom";
import { Grid, Tabs, Tab, Paper, Button, Drawer ,List} from '@material-ui/core';
import React, { useEffect, useState } from 'react';
import ListView from '../adList/ListView';
import Info from '../adInfo/Info';
import NewItemCreation from '../newItemCreation/NewItemCreation';

const Layout = () => {
    const [ tabValue, setTabValue ] = useState(0);
    const [ currentAd, setCurrentAd ] = useState({});
    const [ currentPage, setCurrentPage ] = useState("list");
    const [ isNewDrawer, setIsNewDrawer ] = useState(false);

    useEffect(() => {
        if(currentAd.title){
            setCurrentPage("info");
        }else{
            setCurrentPage("list");
        }
    }, [currentAd]);
    
    return(
        <Grid data-testid="layout">
            {/* New advertisement adding block */}
            <Drawer 
                anchor="right" 
                open={isNewDrawer} 
                // onClose={() => { setIsNewDrawer(false) }}
                >
                <NewItemCreation
                    closeAction={ () => { setIsNewDrawer(false) } }
                />
            </Drawer>

            {/* Header */}
            <Paper className="headerBlock">
                <Grid container display="flex" justifyContent="space-between" alignItems="center">
                    <Grid className="appTitle">Classified Ads</Grid>
                    <Tabs
                        value={ tabValue }
                        indicatorColor="primary"
                        // textColor="primary"
                        onChange={ (e, currentTab) => { setTabValue(currentTab); }}
                    >
                        <Tab label="Home" />
                        <Tab label="Favourites" />
                    </Tabs>
                </Grid>
            </Paper>
            <Grid className="innerPage">
                {/* all the advertisement list */}
                {
                    currentPage === "list" &&
                    <ListView
                        currentPageId={ tabValue }
                        adClickAction={ (item) => { setCurrentAd(item) } }
                        newBtnAction={ () => { setIsNewDrawer(true) } }
                    />
                }
                {
                    // single add - info
                    currentPage === "info" &&
                    <Info
                        ad={currentAd}
                        backClickAction={ () => { setCurrentPage("list"); setCurrentAd({}) } }
                    />
                }
            </Grid>
        </Grid>

    );
}
export default Layout;