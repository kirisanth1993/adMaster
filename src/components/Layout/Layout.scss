.appTitle{
    font-size: 20px;
    font-weight: 500;
    padding-left: 20px;
}

.headerBlock{
    border-radius: 0 !important;
}

.innerPage{
    padding: 20px;
}