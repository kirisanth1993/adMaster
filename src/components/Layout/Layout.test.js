import React from 'react';
import ReactDOM from 'react-dom';
import Layout from './Layout';
import { Provider } from "react-redux";
import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import store from "../../store";

import renderer from "react-test-renderer";

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Provider store={store}><Layout></Layout></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('Layout rendering correctly', () => {
    const {getByTestId} = render(<Provider store={store}><Layout></Layout></Provider>);
    expect(getByTestId('layout')).toHaveTextContent("Classified Ads")
});

it('Layout rendering correctly', () => {
    const {getByTestId} = render(<Provider store={store}><Layout></Layout></Provider>);
    expect(getByTestId('layout')).toHaveTextContent("Home")
});

it("it matches snapshot", () => {
    const tree = renderer.create(<Provider store={store}><Layout></Layout></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})