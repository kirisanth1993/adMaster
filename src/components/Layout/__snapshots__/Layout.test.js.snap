// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div
  className="MuiGrid-root"
  data-testid="layout"
>
  <div
    className="MuiPaper-root headerBlock MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
      display="flex"
    >
      <div
        className="MuiGrid-root appTitle"
      >
        Classified Ads
      </div>
      <div
        className="MuiTabs-root"
      >
        <div
          className="MuiTabs-scroller MuiTabs-fixed"
          onScroll={[Function]}
          style={
            Object {
              "marginBottom": null,
              "overflow": "hidden",
            }
          }
        >
          <div
            className="MuiTabs-flexContainer"
            onKeyDown={[Function]}
            role="tablist"
          >
            <button
              aria-selected={true}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                Home
              </span>
              <span
                className="PrivateTabIndicator-root-40 PrivateTabIndicator-colorPrimary-41 MuiTabs-indicator"
                style={Object {}}
              />
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                Favourites
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root innerPage"
  >
    <div
      className="MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
              id="demo-simple-select-outlined-label"
            >
              Category
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="Category"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="Category"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                All
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value="all"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-44 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-46 PrivateNotchedOutline-legendNotched-47"
                >
                  <span>
                    Category
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-10"
        >
          <div
            className="MuiBox-root MuiBox-root-48"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained new-btn MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                New Classified
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root advertisement-list-block MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-content-xs-space-between"
        display="flex"
      >
        <div
          className="MuiGrid-root nodata MuiGrid-container MuiGrid-justify-content-xs-center"
          display="flex"
        >
          No Data Found
        </div>
      </div>
    </div>
  </div>
</div>
`;
